drivers/nrf_802154_sl/src/platform/clock/nrf_802154_clock_zephyr.c

```diff
@@ -0,0 +1,129 @@
+/* Copyright (c) 2019, Nordic Semiconductor ASA
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   1. Redistributions of source code must retain the above copyright notice, this
+ *      list of conditions and the following disclaimer.
+ *
+ *   2. Redistributions in binary form must reproduce the above copyright notice,
+ *      this list of conditions and the following disclaimer in the documentation
+ *      and/or other materials provided with the distribution.
+ *
+ *   3. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *      contributors may be used to endorse or promote products derived from
+ *      this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/**
+ * @file
+ *   This file implements the nrf 802.15.4 HF Clock abstraction with Zephyr API.
+ *
+ * This implementation uses Zephyr API for clock management.
+ */
+
+#include "platform/clock/nrf_802154_clock.h"
+
+#include <stddef.h>
+
+#include <compiler_abstraction.h>
+#include <drivers/clock_control/nrf_clock_control.h>
+#include <drivers/clock_control.h>
+
+static bool hfclk_is_running;
+static bool lfclk_is_running;
+
+void nrf_802154_clock_init(void)
+{
+    /* Intentionally empty. */
+}
+
+void nrf_802154_clock_deinit(void)
+{
+    /* Intentionally empty. */
+}
+
+void nrf_802154_clock_hfclk_start(void)
+{
+    struct device *clk;
+
+    clk = device_get_binding(DT_INST_0_NORDIC_NRF_CLOCK_LABEL);
+    __ASSERT_NO_MSG(clk != NULL);
+
+    clock_control_on(clk, CLOCK_CONTROL_NRF_SUBSYS_HF); /* Blocking call. */
```

Comment:
You're using some old version of this driver, it's been moved to async API some time ago:
https://github.com/nrfconnect/sdk-hal_nordic/blob/master/drivers/nrf_radio_802154/platform/clock/nrf_802154_clock_zephyr.c

Reply:
@rlubos, should I use the file you mention with this https://github.com/nrfconnect/sdk-nrf/pull/2303#discussion_r429797693 change or without?

Reply:
We should use NCS license in this repository, so yes, please update the license.

Reply:
fixed